src/refda/adm/ietf_dtnma_agent.c

```diff
@@ -1333,13 +1382,183 @@ static void refda_adm_ietf_dtnma_agent_oper_negate(refda_oper_eval_ctx_t *ctx)
+{
+    return left * right;
+}
+static cace_ari_real64 numeric_mul_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left * right;
+}
+
+static cace_ari_uvast numeric_div_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left / right;
+}
+static cace_ari_vast numeric_div_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left / right;
+}
+static cace_ari_real64 numeric_div_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left / right;
+}
+
+static cace_ari_uvast numeric_mod_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left % right;
+}
+static cace_ari_vast numeric_mod_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left % right;
+}
+static cace_ari_real64 numeric_mod_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return fmod(left, right);
+}
+
+static cace_ari_uvast numeric_eq_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left == right;
+}
+static cace_ari_vast numeric_eq_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left == right;
+}
+static cace_ari_real64 numeric_eq_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left == right;
+}
+
+static cace_ari_uvast numeric_ne_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left != right;
+}
+static cace_ari_vast numeric_ne_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left != right;
+}
+static cace_ari_real64 numeric_ne_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left != right;
+}
+
+static cace_ari_uvast numeric_gt_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left > right;
+}
+static cace_ari_vast numeric_gt_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left > right;
+}
+static cace_ari_real64 numeric_gt_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left > right;
+}
+
+static cace_ari_uvast numeric_gte_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left >= right;
+}
+static cace_ari_vast numeric_gte_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left >= right;
+}
+static cace_ari_real64 numeric_gte_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left >= right;
+}
+
+static cace_ari_uvast numeric_lt_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left < right;
+}
+static cace_ari_vast numeric_lt_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left < right;
+}
+static cace_ari_real64 numeric_lt_real64(cace_ari_real64 left, cace_ari_real64 right)
+{
+    return left < right;
+}
+
+static cace_ari_uvast numeric_lte_uvast(cace_ari_uvast left, cace_ari_uvast right)
+{
+    return left <= right;
+}
+static cace_ari_vast numeric_lte_vast(cace_ari_vast left, cace_ari_vast right)
+{
+    return left <= right;
+}
+static cace_ari_real64 numeric_lte_real64(cace_ari_real64 left, cace_ari_real64 right)
```

Comment:
These are verbose in C11 but I think a good pattern to use in the absence of any proper language support.

Reply:
Thanks! I had considered using macros instead but the functions, while verbose, seemed more readable/maintainable.